volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go

```diff
@@ -145,11 +150,85 @@ func (r PersistentVolumeClaimWatcher) removeVolumeFromPvcListAndPvList(logger lo
 	return nil
 }
 
+func (r PersistentVolumeClaimWatcher) addPersistentVolumeClaimToVolumeGroupObjects(
+	logger logr.Logger, pvc *corev1.PersistentVolumeClaim) error {
+	var err error
+	vgList, err := utils.GetVGList(logger, r.Client)
+	if err != nil {
+		return err
+	}
+	err = r.isPVCCanBeAddedToVG(logger, pvc, vgList)
+	if err != nil {
+		return err
+	}
+
+	for _, vg := range vgList.Items {
+		isPVCMatchesVG := false
+		if !utils.IsPVCPartOfVG(pvc, vg.Status.PVCList){
+			isPVCMatchesVG, err = utils.IsPVCMatchesVG(logger, r.Client, pvc, vg)
+			if err != nil {
+				return utils.HandleErrorMessage(logger, r.Client, &vg, err, addingPVC)
+			}
+		}
+
+		if isPVCMatchesVG {
+			err := r.addVolumeFromPvcListAndPvList(logger, pvc, &vg)
+			return utils.HandleErrorMessage(logger, r.Client, &vg, err, addingPVC)
+		}
+	}
+	return nil
+}
+
+func (r PersistentVolumeClaimWatcher) isPVCCanBeAddedToVG(logger logr.Logger, pvc *corev1.PersistentVolumeClaim,
+	vgList csiv1.VolumeGroupList) error {
+	if r.DriverConfig.MultipleVGsToPVC == "true" {
+		return nil
+	}
+	err := utils.IsPVCCanBeAddedToVG(logger, r.Client, pvc, vgList.Items)
+	return utils.HandlePVCErrorMessage(logger, r.Client, pvc, err, addingPVC)
+}
+
+func (r PersistentVolumeClaimWatcher) addVolumeFromPvcListAndPvList(logger logr.Logger,
```

Comment:
add what to what?

Reply:
changed it to `addVolumeToPvcListAndPvList`